virtual_ecosystem/models/animal/functional_group.py

```diff
@@ -129,3 +144,24 @@ def import_functional_groups(
     ]
 
     return functional_group_list
+
+
+def get_functional_group_by_name(
+    functional_groups: tuple[FunctionalGroup, ...], name: str
```

Comment:
I guess the input doesn't *have* to be a tuple...

```suggestion
    functional_groups: Iterable[FunctionalGroup], name: str
```